src/context/authContext/index.jsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { onAuthStateChanged } from "firebase/auth";
+import { firebaseAuth } from "../../firebase/firebase";
+import { useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const AuthContext = React.createContext();
+
+export function useAuth() {
+  return useContext(AuthContext);
+}
+
+export function AuthProvider({ children }) {
+  const [userToken, setUserToken] = useState(null); // Changed from currentUser to userToken
```

Comment:
Can you change this to userInfo, and remove the comment?
```suggestion
  const [userInfo, setUserInfo] = useState(null);
```